jmclient/jmclient/configure.py

```diff
@@ -377,33 +377,18 @@ def get_p2pk_vbyte():
 
 def validate_address(addr):
     try:
-        assert len(addr) > 2
-        if addr[:2].lower() in ['bc', 'tb']:
-            # Regtest special case
-            if addr[:4] == 'bcrt':
-                if btc.bech32addr_decode('bcrt', addr)[1]:
-                    return True, 'address validated'
-                return False, 'Invalid bech32 regtest address'
-            #Else, enforce testnet/mainnet per config
-            if get_network() == "testnet":
-                hrpreq = 'tb'
-            else:
-                hrpreq = 'bc'
-            if btc.bech32addr_decode(hrpreq, addr)[1]:
-                return True, 'address validated'
-            return False, 'Invalid bech32 address'
-        #Not bech32; assume b58 from here
-        ver = btc.get_version_byte(addr)
-    except AssertionError:
-        return False, 'Checksum wrong. Typo in address?'
-    except Exception:
-        return False, "Invalid bitcoin address"
-    if ver != get_p2pk_vbyte() and ver != get_p2sh_vbyte():
-        return False, 'Wrong address version. Testnet/mainnet confused?'
-    if len(btc.b58check_to_bin(addr)) != 20:
-        return False, "Address has correct checksum but wrong length."
-    return True, 'address validated'
-
+        # automatically respects the network
+        # as set in btc.select_chain_params(...)
+        dummyaddr = btc.CCoinAddress(addr)
+    except Exception as e:
+        return False, repr(e)
+    # additional check necessary because python-bitcointx
+    # does not check hash length on p2sh construction.
```

Comment:
This was an omission on python-bitcointx part. Fixed in https://github.com/Simplexum/python-bitcointx/commit/8c106682049f910cd2765b1cc45ba4cbd68636f1

Reply:
Thanks, and sorry for not reporting it. I guess I was deep in the weeds at that point.

Reply:
I would appreciate an issue raised in on python-bitcoinx issue page if something like this is encountered in the future. Thre was no point for bitcointx to ignore and not check the length.

Reply:
Will do.